src/components/application_manager/test/application_manager_impl_test.cc

```diff
@@ -1666,6 +1670,53 @@ TEST_F(ApplicationManagerImplTest,
       app_manager_impl_->RegisterApplication(request_for_registration_ptr);
   EXPECT_EQ(0, application.use_count());
 }
+
+TEST_F(ApplicationManagerImplTest, PolicyIDByIconUrl_Success) {
+  app_manager_impl_->SetMockPolicyHandler(mock_policy_handler_);
+  std::vector<std::string> enabled_apps{"1234"};
+  EXPECT_CALL(*mock_policy_handler_, GetEnabledCloudApps(_))
+      .WillOnce(SetArgReferee<0>(enabled_apps));
+  EXPECT_CALL(*mock_policy_handler_, GetCloudAppParameters(_, _, _, _, _, _, _))
+      .WillOnce(DoAll(SetArgReferee<1>(kEnabled),
+                      SetArgReferee<2>(kEndpoint2),
+                      SetArgReferee<3>(kCertificate),
+                      SetArgReferee<4>(kAuthToken),
+                      SetArgReferee<5>(kTransportType),
+                      SetArgReferee<6>(kHybridAppPreferenceStr),
+                      Return(true)));
+
+  std::vector<std::string> nicknames{"CloudApp"};
+  EXPECT_CALL(*mock_policy_handler_, GetInitialAppData(_, _, _))
+      .WillOnce(DoAll(SetArgPointee<1>(nicknames), Return(true)));
+
+  const std::string url = "https://www.fakeiconurl.com/icon.png";
+  EXPECT_CALL(*mock_policy_handler_, GetIconUrl("1234"))
+      .WillRepeatedly(Return(url));
+
+  app_manager_impl_->RefreshCloudAppInformation();
+
+  std::string result = app_manager_impl_->PolicyIDByIconUrl(url);
+  EXPECT_EQ(result, "1234");
+}
+
+TEST_F(ApplicationManagerImplTest, SetIconFileFromSystemRequest_Success) {
+  CreatePendingApplication();
+  Mock::VerifyAndClearExpectations(mock_message_helper_);
+
+  file_system::CreateDirectory(kDirectoryName);
+  const std::string full_icon_path = kDirectoryName + "/1234";
+  ASSERT_TRUE(file_system::CreateFile(full_icon_path));
+
+  const std::string url = "https://www.fakeiconurl.com/icon.png";
+  EXPECT_CALL(*mock_policy_handler_, GetIconUrl("1234"))
+      .WillRepeatedly(Return(url));
+  EXPECT_CALL(*mock_rpc_service_, ManageHMICommand(_, _)).Times(1);
```

Comment:
Looks like the `UpdateAppList` is verified using this block normally, might be more specific that just `ManageHMICommand`:
```

  smart_objects::SmartObject dummy_object(SmartType_Map);
  SmartObjectSPtr sptr = std::make_shared<SmartObject>(dummy_object);

  EXPECT_CALL(*mock_message_helper_,
              CreateModuleInfoSO(
                  hmi_apis::FunctionID::BasicCommunication_UpdateAppList, _))
      .WillOnce(Return(sptr));
  EXPECT_CALL(*mock_rpc_service_, ManageHMICommand(sptr, _)).Times(1);
```